bblocks/import_tools/world_bank.py

```diff
@@ -231,98 +215,151 @@ def _clean_pink_sheet(df: pd.DataFrame, sheet: str) -> pd.DataFrame:
+    when `load_indicator` is called for the first time.
+    You can force an update by calling `update` if you want to refresh the data stored on disk.
+    You can get a dataframe of the data by calling `get_data`
 
-    Parameters:
-        sheet (str): name of the sheet in the Pink Sheet ['Monthly Prices', 'Monthly Indices']
     """
 
-    sheet: str
-    data: pd.DataFrame = None
-    update_data: bool = False
+    sheet: str = None
 
-    def __post_init__(self) -> None:
-        if self.update_data:
-            self.update()
-        if not os.path.exists(f"{PATHS.imported_data}/{self.file_name}"):
+    def __post_init__(self):
+        if self.sheet not in ["Monthly Indices", "Monthly Prices"]:
+            raise ValueError(
+                "Invalid sheet name. "
+                "Please specify 'Monthly Indices' or 'Monthly Prices'"
+            )
+        self.indicators = {}
+
+    def __load(self, indicator_list: list) -> None:
+        """Loads indicators and data from disk
+
+        Args:
+            indicator_list: list of indicators to load"""
+
+        df = pd.read_csv(f"{PATHS.imported_data}/{self.file_name}")
+
+        if indicator_list == "all":
+            indicator_list = list(df.indicator.unique())
+
+        elif isinstance(indicator_list, str):
+            indicator_list = [indicator_list]
+
+        for indicator in indicator_list:
+            if indicator not in df.indicator.unique():
+                raise Warning(f"{indicator} not found")
+            else:
+                self.indicators[indicator] = df[df.indicator == indicator].reset_index(
+                    drop=True
+                )
+
+        self.data = df[df.indicator.isin(list(self.indicators))].reset_index(drop=True)
+
+    def load_indicator(self, indicator: Optional | str | list = "all") -> ImportData:
+        """Load data for an indicator or list of indicators.
+
+        Args:
+            indicator: indicator to load, either 'holdings' or 'allocations'. The default
+                is 'all' which loads both.
+
+        Returns:
+            The same object to allow chaining
+        """
+
+        if not os.path.exists(f"{PATHS.imported_data}/{self.file_name}") or (
+            self.update_data and self.data is None
+        ):
             self.update()
-        else:
-            self.data = pd.read_csv(f"{PATHS.imported_data}/{self.file_name}")
 
-    @property
-    def file_name(self) -> str:
-        """Returns the name of the stored file"""
-        return f"World Bank Pink Sheet - {self.sheet}.csv"
+        self.__load(indicator)
+
+        return self
+
+    def update(self, reload_data=False) -> ImportData:
+        """Update the data saved on disk
+
+        When called it downloads Pink sheet Data from the World Bank and saves it to disk.
+        Optionally specify whether to reload the data to the object
+
+        Args:
+            reload_data: If True, the data will be reloaded to the object
```

Comment:
Nice solution!

Reply:
wondering if this should be set to True as default @jm-rivera ?

Reply:
Yes I think that would be a good idea